### DISCUSSION TITLE
Custom actions

### DISCUSSION THREAD

**Discussion by elijaholmos** [stalniy/casl#780] [Q&A] [answered]
First, is there any statically typed list of actions exported by CASL? I couldn't find anything on the documentation, and it seems to be implicit that the list of actions is as follows:
```ts
type Actions = 'create' | 'read' | 'update' | 'delete' | 'manage';
```

Second, am I able to create my own actions? The specific use case I am thinking of is to use the verb "remove" instead of "delete". I have some pre-existing automation & code inference that I am integrating with, so this is not simply a matter of preference. Again, couldn't find anything in the documentation.

**Comment by jd4u** [comment, marked as answer]
First: No.
Second: Yes. You can create your typed actions and use them in creating rules as well as checking them.

**Reply by elijaholmos** [reply]
Yeah, as I have discovered, we make our own actions and use that in ability creation. The ones provided in the documentation are merely what could be considered common or conventional. Thank you
